cmds/dhclient/dhclient.go

```diff
@@ -247,12 +281,17 @@ func main() {
 			} else {
 				done <- dhclient6(iface, *renewals, timeout, *retry)
 			}
-		}(i)
+			debug("Done dhclient for %v", ifname)
+		}(i.Attrs().Name)
 	}
 
+	go func() {
+		wg.Wait()
```

Comment:
You don't need the WaitGroup because of the blocking, unbuffered done channel. I mean, this approach definitely works, too, but it's an additional goroutine and additional WaitGroup that you don't actually need.